tests/scripts/all.sh

```diff
@@ -2646,6 +2646,55 @@ component_test_psa_crypto_config_accel_all_curves_except_x25519 () {
     psa_crypto_config_accel_all_curves_except_one MBEDTLS_ECP_DP_CURVE25519_ENABLED
 }
 
+# This is an helper used by:
+# - component_test_psa_ecc_key_pair_no_derive
+# - component_test_psa_ecc_key_pair_no_generate
+# The goal is to test with all PSA_WANT_KEY_TYPE_xxx_KEY_PAIR_yyy symbols
+# enabled, but one. Input arguments are as follows:
+# - $1 is the key type under test, i.e. ECC/RSA/DH
+# - $2 is the key option to be unset (i.e. generate, derive, etc)
+config_test_psa_want_key_pair_partial() {
+    KEY_TYPE=$1
+    UNSET_OPTION=$2
+
+    msg "build: full + MBEDTLS_PSA_CRYPTO_CONFIG + PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_xxx"
+    scripts/config.py full
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
+    scripts/config.py unset MBEDTLS_SSL_PROTO_TLS1_3
```

Comment:
In terms of what's important, we really need to have a test with RSA generation turned off, because that's a big driver of code size and it's not needed on devices that only import private keys. DH and ECC generation requires only a small amount of code and is usually needed (though not if you're only ever doing static DH), so it's a lot less important if turning those off doesn't work.

But from a systematic testing point of view it does make sense to disable generation for everything. So given a choice of: (disable all generation and not test TLS1.3) vs (disable only RSA generation and test TLS 1.3), I think it's a wash. But disabling USE_PSA_CRYPTO is less clear: users who use the fancy PSA stuff are likely to turn that on.

Reply:
Note: as it currently stands, the function does this for an individual key type, not all 3 of them at once. From a testing perspective, I think that's good: for example if ECDH key generation is disabled but FFDH generate is enabled, we won't be able to do an ECDHE-ECDSA key exchange with TLS 1.2 with `USE_PSA` but we should be able to do a DHE-RSA key exchange. Similarly, with TLS 1.3, if only one of ECDH or FFDH generate is available, we should still be able to do an ephemeral key exchange and automatically negociate the kind of group that we can actually use. (However, that last part is not true yet, as (1) FFDH support in TLS 1.3 is not merged yet, and (2) it doesn't have proper check for `GENERATE`, which we'll probably want to add later, perhaps as part of #7612 where we clean up other dependencies related to ECDH in TLS.)

Now that I've said that, I think I definitely want to test with `USE_PSA` and TLS 1.3 at some point. Probably not right now, as I guess we'll need to fix a few guards in TLS first (again, can be part of #7612), but in the end.

Reply:
TBH when I was working for introducing this test I found that there was already a component, `component_build_psa_accel_key_type_ecc_key_pair`, which seem to do what I was looking for so I basically started/copied from there.
That's was mostly an initial attempt, so I'll try to address your comments ;)